EssayAnalyzer.UnitTest/Services/Foundation/Users/UserServiceTests.Exception.Add.cs

```diff
@@ -161,4 +161,37 @@ public async Task ShouldThrowDependencyValidationExceptionOnAddIfDatabaseUpdateE
         this.storageBrokerMock.VerifyNoOtherCalls();
         this.loggingBrokerMock.VerifyNoOtherCalls();
     }
+
+    [Fact]
+    public async Task ShouldThrowServiceExceptionOnAddIfServiceErrorOccursAndLogItAsync()
+    {
+        // given
+        User randomUser = CreateRandomUser();
+        Exception serviceException = new Exception();
+
+        var failedUserServiceException = 
+            new FailedUserServiceException(serviceException);
+
+        var expectedUserServiceException = 
+            new UserServiceException(failedUserServiceException);
+
+        this.storageBrokerMock.Setup(broker =>
+            broker.InsertUserAsync(randomUser)).ThrowsAsync(serviceException);
+        
+        // when
+        ValueTask<User> addUserTask = this.userService.AddUserAsync(randomUser);
+
+        UserServiceException actualUserServiceException =
+            await Assert.ThrowsAsync<UserServiceException>(addUserTask.AsTask);
+
+        // then
+        actualUserServiceException.Should().BeEquivalentTo(expectedUserServiceException);
```

Comment:
remove this line

Reply:
this line for seperating // then and // when